src/com/tekion/cricket/Match.java

```diff
@@ -0,0 +1,267 @@
+                 team1.incrementTeamScore(5);
+                 team1.changeStrike();
+                 break;
+             case 6:
+                 System.out.println(bowler.getName()+" to "+batsmen.getName()+" "+result);
+                 team1.incrementTeamScore(6);
+                 break;
+             case 7:
+                 System.out.println(bowler.getName()+" to "+batsmen.getName()+" Out");
+                 handleWicket(team1,team2);
+                 break;
+
+         }
+
+    }
+
+    void simulateFirstInning(Team team1, Team team2)
+    {
+        int overs = totalAvailableBalls/6;
+        team1.setStrikerPlayer(0);
+        team1.setNonStrikerPlayer(1);
+        team1.setNextPlayer(2);
+        int currentBowler=5;
+        for(int over=0;over<overs;over++)
+        {
+            team2.setCurrentBowler(currentBowler);
+            System.out.println("Over Number: "+(over+1));
+            System.out.println("-------------------------");
+
+            for(int ball=1;ball<=6;ball++)
+            {
+                if(hasLastWicketFallen(team1))
+                    return;
+
+                team1.incrementTotalPlayedBalls();
+                team2.manageBowlersBalls();
+
+                playBall(team1,team2);
+
+            }
+            System.out.println("-------------------------------");
+            team1.changeStrike();
+            if(currentBowler==10)currentBowler=4;
+            currentBowler++;
+        }
+    }
+
+    void simulateSecondInning(Team team1, Team team2, int scoreToChase)
+    {
+        int overs = totalAvailableBalls/6;
+        team2.setStrikerPlayer(0);
+        team2.setNonStrikerPlayer(1);
+        team2.setNextPlayer(2);
+        int currentBowler=5;
+        for(int over=0;over<overs;over++)
+        {
+            team1.setCurrentBowler(currentBowler);
+            System.out.println("Over Number: "+(over+1));
+            System.out.println("-------------------------");
+            for(int ball=1;ball<=6;ball++)
+            {
+                if(hasLastWicketFallen(team2) || scoreToChase < team2.getTeamScore())
+                    return;
+
+                team2.incrementTotalPlayedBalls();
+                team1.manageBowlersBalls();
+
+                playBall(team2,team1);
+
+            }
+            System.out.println("-------------------------------");
+            team2.changeStrike();
+            if(currentBowler==10)currentBowler=4;
+            currentBowler++;
+        }
+    }
+
+    void declareWinner(Team team1, Team team2)
+    {
+        if(team1.getTeamScore() > team2.getTeamScore())
+        {
+            winner = Winner.TEAM1;
+        }
+        else if(team1.getTeamScore() < team2.getTeamScore())
+        {
+            winner = Winner.TEAM2;
+        }
+        else if(team1.getTeamScore() == team2.getTeamScore())
+        {
+            winner = Winner.TIE;
+        }
+        else{
+            winner = Winner.DRAW;
+        }
+    }
+
+    void startMatch()
```

Comment:
can we optimise this method further by moving out some of the lines to another method ?

Reply:
Yes sir, I'll push the modified version in the week-2 branch.